gms/version_generator.cc

```diff
@@ -8,15 +8,27 @@
  * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
  */
 
+#include <exception>
+
+#include <fmt/format.h>
+
 #include "version_generator.hh"
 
 namespace gms {
+
+void validate_gossip_generation(int64_t generation_number) {
+    if (generation_number > std::numeric_limits<gms::generation_type::value_type>::max() ||
+        generation_number < std::numeric_limits<gms::generation_type::value_type>::min()) {
+        throw std::out_of_range(fmt::format("gossip generation {} is out of range", generation_number));
+    }
```

Comment:
nit, could use `std::in_range<gms::generation_type::value_type>(generation_number)` here.

Reply:
done in v5